### PR TITLE
(fix): update usage of `AxisArrays` and related classes

### LINKED ISSUE #77: MuData is incompatible with AnnData v0.10.9
Import broken with AnnData version 0.10.9

**System**
 - OS: Ubuntu
 - 3.9
 -AnnData 0.10.9

```
  File "/home/cane/.local/share/hatch/env/virtual/scvi-tools/E5IuCX_W/scvi-tools/lib/python3.12/site-packages/mudata/_core/mudata.py", line 18, in <module>
    from anndata._core.aligned_mapping import (
ImportError: cannot import name 'AlignedViewMixin' from 'anndata._core.aligned_mapping' (/home/cane/.local/share/hatch/env/virtual/scvi-tools/E5IuCX_W/scvi-tools/lib/python3.12/site-packages/anndata/_core/aligned_mapping.py). Did you mean: 'AlignedView'?
```

### PR DESCRIPTION
**Pull request by ilan-gold**
Fixes #77

### CODE REVIEW

**Comment by ilan-gold** [comment]
@gtca

File: src/mudata/_core/file_backing.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ def __init__(
         self._children = WeakSet()
         if filename is not None:
             filename = Path(filename)
-        super().__init__(None, filename, filemode)
+        super().__init__(ad.AnnData(), filename, filemode)
```
Comment (ilan-gold) [inline src/mudata/_core/file_backing.py:21]: This is the one I'm least sure of tbh.  Seems like `None` and `ad.AnnData` are just as good (bad) here but 🤷

**Comment by ilan-gold** [comment]
Also I think you may be missing test coverage @gtca : https://github.com/scverse/mudata/pull/78/commits/907f8f1bf67807cb5a5a6b4a2914c826ac9a801f probably should have errored

**Comment by gtca** [comment]
@ilan-gold we could use improved test coverage indeed...

**Pull request merged**